From 2bbbc56b21d033c5f3fa5d23fbe18a0b82060de1 Mon Sep 17 00:00:00 2001
From: Jeff Morriss <jeff.morriss.ws@gmail.com>
Date: Wed, 13 Aug 2014 21:05:38 -0400
Subject: [PATCH 4/6] Fix the uninitialized-read error reported in bug 10370.

Fix apparent cut-n-pasteo: if offset_e is set then we should be looking
at offset_e (instead of offset_s) and ts_end (instead of ts_start).

Bug: 10370
Change-Id: I0683ad55ac782c7fd03b7dbdf3f34df713267de5
Reviewed-on: https://code.wireshark.org/review/3589
Reviewed-by: Evan Huus <eapache@gmail.com>
(cherry picked from commit 44698259b1f5865c60323acaf2a633654a2abe81)
Reviewed-on: https://code.wireshark.org/review/3591
(cherry picked from commit 2035d31c0e7cb7291ede2544a04b4641dc9b12d7)
Reviewed-on: https://code.wireshark.org/review/3592
Reviewed-on: https://code.wireshark.org/review/4425
Reviewed-by: Balint Reczey <balint@balintreczey.hu>
Tested-by: Balint Reczey <balint@balintreczey.hu>
---
 epan/dissectors/packet-netflow.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/epan/dissectors/packet-netflow.c b/epan/dissectors/packet-netflow.c
index 026a164..1f8d98e 100644
--- a/epan/dissectors/packet-netflow.c
+++ b/epan/dissectors/packet-netflow.c
@@ -5110,7 +5110,7 @@ dissect_v9_v10_pdu_data(tvbuff_t *tvb, packet_info *pinfo, proto_tree *pdutree,
                                         offset_e[i], 4, &ts_end[i]);
                 } else {
                     proto_tree_add_time(pdutree, hf_cflow_abstimeend, tvb,
-                                        offset_s[i], 4, &ts_start[i]);
+                                        offset_e[i], 4, &ts_end[i]);
                 }
             }
         }
-- 
2.1.1

